internal/security/secretstore/tokenfilewriter/tokenfilewriter_test.go

```diff
@@ -0,0 +1,189 @@
+			secretClient := &mocks.SecretStoreClient{}
+
+			secretClient.On("InstallPolicy", testRootToken,
+				consulSecretsEngineOpsPolicyName, mock.Anything).
+				Return(test.installPolicyError).Once()
+
+			secretClient.On("CreateToken", testRootToken, mock.Anything).
+				Return(createTokenResult, test.createTokenError).Once()
+
+			token, _, err := NewWriter(lc, secretClient, flOpener).
+				CreateMgmtTokenForConsulSecretsEngine(testRootToken)
+
+			if test.installPolicyError != nil || test.createTokenError != nil {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, "createdTestToken", token["auth"].(map[string]interface{})["client_token"])
+				secretClient.AssertExpectations(t)
+			}
+		})
+	}
+}
+
+func TestCreateAndWriteForConsulSecretEngine(t *testing.T) {
+	lc, flOpener := setup()()
+
+	createTokenResult := getCreateTokenResultStub()
+
+	testInstallPolicyErr := errors.New("install policy error")
+	testCreateTokenErr := errors.New("create token error")
+	testTokenFileDir := "test"
+	testRootToken := "testRootToken"
+
+	tests := []struct {
+		name                 string
+		rootToken            string
+		installPolicyCallErr error
+		createTokenCallErr   error
+	}{
+		{"Ok:CreateAndWrite with client call ok", testRootToken, nil, nil},
+		{"Bad:CreateAndWrite with empty token", "", nil, nil},
+		{"Bad:CreateAndWrite with install policy call error", testRootToken, testInstallPolicyErr, nil},
+		{"Bad:CreateAndWrite with create token call error", testRootToken, nil, testCreateTokenErr},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// setup mock secretclient and expected return values for this test
+			secretClient := &mocks.SecretStoreClient{}
+
+			secretClient.On("InstallPolicy", testRootToken,
+				consulSecretsEngineOpsPolicyName, mock.Anything).
+				Return(test.installPolicyCallErr).Once()
+
+			secretClient.On("CreateToken", testRootToken, mock.Anything).
+				Return(createTokenResult, test.createTokenCallErr).Once()
+
+			testTokenFilePath := filepath.Join(testTokenFileDir,
+				"mgmt-token.json"+"_"+strconv.FormatInt(time.Now().UnixNano(), 10))
+			fileWriter := NewWriter(lc, secretClient, flOpener)
+			_, err := fileWriter.CreateAndWrite(test.rootToken, testTokenFilePath,
+				fileWriter.CreateMgmtTokenForConsulSecretsEngine)
+			defer func() {
+				_ = os.RemoveAll(filepath.Dir(testTokenFilePath))
+			}()
+
+			expectError := test.installPolicyCallErr != nil ||
+				test.createTokenCallErr != nil ||
+				len(test.rootToken) == 0
+
+			if expectError {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.FileExists(t, testTokenFilePath)
+			}
+		})
+	}
+}
+
+func TestFileWriteErrorForConsulSecretEngine(t *testing.T) {
+	curUser, _ := user.Current()
+	if curUser != nil && curUser.Uid == "0" {
+		// it is root user then we skip this test as root can have permission to write everything
+		t.Skip()
```

Comment:
Should we fail rather than skip so developer is forced to not run the tests as root?

Reply:
not sure, if CI is running as root then the test will be failing. Is CI running as root user?

Reply:
I doubt it, but @ernestojeda should be able to answer the question.

Reply:
Yes, when `make test` runs, it runs inside the context of a docker image and we run as root due to permissions issues with the LF build agents. See: https://github.com/edgexfoundry/edgex-global-pipelines/blob/master/vars/edgeXBuildGoApp.groovy#L181

Reply:
Thank you, @ernestojeda.  @lenny-intel i'll see this comment as resolved then.

Reply:
See my other comment about mocking the file creation commands and thus having to avoid this debacle althogether.